File: core-java-practice/leet-code-codebase/AssignCookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int a = g.length-1;
        int e = s.length-1;
        int re =0;
        while(e>=0&&a>=0){
            if(g[a] <=s[e]){
                re+=1;
                e--;
            }
            a--;
        }
        return re;
    }
}

File: core-csharp-practice/leet-code-codebase/AssignCookies.cs
using System;

class Solution
{
    public int FindContentChildren(int[] g, int[] s)
    {
        Array.Sort(g);
        Array.Sort(s);

        int a = g.Length - 1;
        int e = s.Length - 1;
        int re = 0;

        while (e >= 0 && a >= 0)
        {
            if (g[a] <= s[e])
            {
                re++;
                e--;
            }
            a--;
        }

        return re;
    }
}
